esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,234 @@
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368
+        <https://github.com/SciTools-incubator/iris-esmf-regrid/issues/368>`_
+        has been resolved.
+    use_tgt_mask:
+        If True, derive a mask from of the target cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368`_ has been
+        resolved.
+    collapse_src_mask_along:
+        When deriving the mask from the source cube data, collapse the mask
+        along the dimensions identified by these axes or coordinates. Only
+        points that are masked at all time (``'T'``), vertical levels
+        (``'Z'``), or both time and vertical levels (``'TZ'``) will be
+        considered masked. Instead of the axes ``'T'`` and ``'Z'``,
+        coordinate names can also be provided. For any cube dimensions not
+        specified here, the first slice along the coordinate will be used to
+        determine the mask.
+    collapse_tgt_mask_along:
+        When deriving the mask from the target cube data, collapse the mask
+        along the dimensions identified by these axes or coordinates. Only
+        points that are masked at all time (``'T'``), vertical levels
+        (``'Z'``), or both time and vertical levels (``'TZ'``) will be
+        considered masked. Instead of the axes ``'T'`` and ``'Z'``,
+        coordinate names can also be provided. For any cube dimensions not
+        specified here, the first slice along the coordinate will be used to
+        determine the mask.
+    src_resolution:
+        If present, represents the amount of latitude slices per source cell
+        given to ESMF for calculation. If resolution is set, the source cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_resolution:
+        If present, represents the amount of latitude slices per target cell
+        given to ESMF for calculation. If resolution is set, the target cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_location:
+        Only used if the target grid is an :class:`iris.mesh.MeshXY`. Describes
```

Comment:
```suggestion
        Only used if the target grid is an :class:`iris.mesh.MeshXY` or an :class:`iris.cube.Cube` with an associated mesh. Describes
```

Reply:
It is not used if the target is a cube (on a mesh), see the original docstring: https://iris-esmf-regrid.readthedocs.io/en/latest/_api_generated/esmf_regrid.schemes.html#esmf_regrid.schemes.ESMFBilinear

In that case, the `location` attribute of the cube is used: https://github.com/SciTools-incubator/iris-esmf-regrid/blob/9e2559af148b0e5ac057fcfeacac3a7ec3e18739/esmf_regrid/schemes.py#L504

Reply:
Good point, haven't checked the code in that detail 😄